felix/cmd/calico-bpf/commands/counters.go

```diff
@@ -99,11 +99,25 @@ func dumpInterface(cmd *cobra.Command, iface string) error {
 	if iface == "" {
 		return fmt.Errorf("empty interface name")
 	}
-
 	bpfCounters := counters.NewCounters(iface)
-	values, err := bpfCounters.Read()
-	if err != nil {
-		return fmt.Errorf("Failed to read bpf counters. iface=%s err=%v", iface, err)
+
+	values := make([][]uint64, len(bpf.Hooks))
+	for i := range values {
+		values[i] = make([]uint64, counters.MaxCounterNumber)
+	}
```

Comment:
why this allocation?

Reply:
`make` only creates the `[][]byte' but each entry `[]byte` is still `nil` thus accessing it panics. It's seems mandatory to also init each of `[]byte` in the 2D `[][]byte`.

Reply:
But it is then assigned `values[index] = val` sure you would need to check if it is set at `newRow = append(newRow, fmt.Sprintf("%v", values[index][c.Counter]))` but isn't that likely what you want anyway when say the XDP counters are missing?